File: javascript/core/src/verbs/ungroup.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { TableStep } from './nodeFactories/index.js'
import { stepNodeFactory } from './nodeFactories/StepNode.js'

export const ungroupStep: TableStep<void> = input => input.ungroup()
export const ungroup = stepNodeFactory(ungroupStep)

File: python/datashaper/datashaper/verbs/ungroup.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Ungroup verb implementation."""

import pandas as pd
from reactivedataflow import InputPort, verb

from datashaper import DEFAULT_INPUT_NAME

from .decorators import OutputMode, wrap_verb_result


@verb(
    name="ungroup",
    ports=[
        InputPort(name=DEFAULT_INPUT_NAME, parameter="table", required=True),
    ],
    adapters=[
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def ungroup(table: pd.DataFrame) -> pd.DataFrame:
    """Ungroup verb implementation."""
    return table.obj
